test/jdk/javax/swing/JTableHeader/8016524/JTHeaderBorderTest.java

```diff
@@ -0,0 +1,120 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8016524
+ * @requires (os.family=="mac")
+ * @key headful
+ * @summary Tests whether the bottom line of JTableHeader border is visible for MacOS default LAF
+ * @run main JTHeaderBorderTest
+ */
+
+import java.awt.*;
+import javax.imageio.ImageIO;
+import javax.swing.JFrame;
+import javax.swing.JTable;
+import javax.swing.JScrollPane;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB;
+
+public class JTHeaderBorderTest {
+
+    private static JFrame frame;
+    private static JTable table;
+    private static JScrollPane scrollableTable;
+
+    private static final int FRAME_HT = 300;
+    private static final int FRAME_WT = 300;
+    private static final int TABLE_COLS = 3;
+    private static final int TABLE_ROWS = 2;
+    private static final int Y_OFFSET = 30;
+    private static final int X_OFFSET = 25;
+
+    public static void main(String[] args) throws Exception {
+
+        try {
+            try {
+                UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException
+                    | UnsupportedLookAndFeelException e) {
+                throw new RuntimeException("Unsupported Look&Feel Class");
+            }
+
+            SwingUtilities.invokeAndWait(() -> {
+                table = new JTable(TABLE_ROWS, TABLE_COLS);
+                scrollableTable = new JScrollPane(table);
+
+                frame = new JFrame();
+                frame.getContentPane().add(scrollableTable);
+                frame.setSize(FRAME_WT, FRAME_HT);
+                frame.setLocationRelativeTo(null);
+                frame.setVisible(true);
+                frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+                // paint JFrame to BufferedImage
+                BufferedImage image = new BufferedImage(FRAME_WT, FRAME_HT, TYPE_INT_ARGB);
+                Graphics2D graphics2D = image.createGraphics();
+                frame.paint(graphics2D);
+                graphics2D.dispose();
+
+                String tableColor = Integer.toHexString(table.getBackground().getRGB());
+                String headerColor = Integer.toHexString(table.getTableHeader().getBackground().getRGB());
+                String pixelColor = tableColor;
+                boolean isBottomLineVisible = false;
+
+                // scan table header region to check if bottom border of JTableHeader is visible
+                for (int y = Y_OFFSET; y <= Y_OFFSET+25; y++) {
```

Comment:
The X_OFFSET is different from the 25 specified in the loop, this specifies the vertical scan range. To avoid confusion I have changed it as Y_OFFSET_START and Y_OFFSET_END.